db/migrate/20170519153350_add_showback_models.rb

```diff
@@ -0,0 +1,73 @@
+class AddShowbackModels < ActiveRecord::Migration[5.0]
+  def up
+    create_table :showback_usage_types do |t|
+      t.string     :category
+      t.string     :description
+      t.string     :measure
+      t.text       :dimensions
+      t.timestamps
+    end
+
+    create_table :showback_events do |t|
+      t.jsonb      :data
+      t.timestamp  :start_time
+      t.timestamp  :end_time
+      t.belongs_to :resource, :type => :bigint, :polymorphic => true, :index => true
+      t.jsonb      :context
+      t.timestamps
+    end
+
+    create_table :showback_price_plans do |t|
+      t.string     :name
+      t.belongs_to :resource, :type => :bigint, :polymorphic => true
+      t.string     :description
+      t.timestamps
+    end
+
+    create_table :showback_rates do |t|
+      t.bigint     :fixed_rate_cents # Columns needed by gem money
+      t.string     :fixed_rate_currency
+      t.bigint     :variable_rate_cents # Columns needed by gem money
+      t.string     :variable_rate_currency
+      t.string     :calculation
+      t.string     :category
+      t.string     :dimension
+      t.jsonb      :screener
+      t.datetime   :date
+      t.string     :concept
+      t.belongs_to :showback_price_plan, :type => :bigint
+      t.timestamps
+    end
+    add_index :showback_rates, :screener, :using => :gin
```

Comment:
It is jsonb because it is independent on the content. This is key element here: I don't care what is in chargeback, our rating is independent on anything that is inside. Using columns will make again the relationship manual, and thus I won't have what I want. With this schema I can charge anything (i.e. any data, any context, any tag, anything), with the same function. Adding a new charge is a matter of adding a new event and nothing else, I don't need to modify rating because I want to charge another event. If I do that with columns, then I need to do a migration every time I want to include a new event or system. (i.e. networking, storage, containers, pods, dockerfiles, you name it)

We don't want columns.
- We would need to define columns every time we want to rate, and that would not work.
- I want to use screener to match against context in a generic way: {screener} should be included in {context} to be used. I don't know before hand what is in screener or what is in context in the showback_event (i.e. tag: silver in context should have to be in screener to be used). More than that, I don0't care, that is for the customer to define, not me.
- In the way defined right now, it is quite easy to get that content into a non-sql database and use it directly.